DockerfileGPU

```diff
@@ -2,7 +2,7 @@ FROM unifyai/ivy:latest-gpu
 
 # Install Ivy
 RUN rm -rf ivy && \
-    git clone https://github.com/unifyai/ivy && \
+    git clone --recurse-submodules https://github.com/unifyai/ivy && \
```

Comment:
while this is fine, I don't think `--recurse-submodules` is actually strictly necessary. The only submodule is the Array API test suite, which we don't make use of there. I think it's probably simpler to remove this for slightly faster docker image building.